factory.cpp

```diff
@@ -0,0 +1,67 @@
+#include <iostream>
+using namespace std;
+ 
+class Bicycle {
+public:
+    virtual void printBicycle() = 0;
+};
+
+class FourWheel : public Bicycle {
```

Comment:
Lets use more descriptive class names. e.g. `FourWellBicycle` and `TwoWheelBicycle`.